test/Garnet.test/Extensions/RateLimiterTxn.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Text;
+using Garnet.common;
+using Garnet.server;
+using Tsavorite.core;
+
+namespace Garnet
+{
+    sealed class RateLimiterTxn : CustomTransactionProcedure
+    {
+        public override bool Prepare<TGarnetReadApi>(TGarnetReadApi api, ref CustomProcedureInput procInput)
+        {
+            int offset = 0;
+            AddKey(GetNextArg(ref procInput, ref offset), LockType.Exclusive, true);
+            return true;
+        }
+
+        public override unsafe void Main<TGarnetApi>(TGarnetApi api, ref CustomProcedureInput procInput, ref MemoryResult<byte> output)
```

Comment:
Do we need a way for measuring latency and KPS metrics for these custom transactions?

Reply:
TX_PROC_LAT helps track latency for custom transactions.